libs/src/api/entityManager.ts

```diff
@@ -128,14 +129,83 @@ export class EntityManager extends Base {
       })
       logger.info(`DeletePlaylistData - ${JSON.stringify(resp)}`)
       const txReceipt = resp.txReceipt
-      respValues = {
+      return {
         blockHash: txReceipt.blockHash,
         blockNumber: txReceipt.blockNumber
       }
     } catch (e) {
       logger.error(`Data delete playlist: err ${e}`)
+      throw e
+    }
+  }
+  /**
+   * Update a playlist using updated data contracts flow
+   **/
+  async updatePlaylist({
+    playlistId,
+    playlistName,
+    trackIds,
+    description,
+    isAlbum,
+    isPrivate,
+    coverArt,
+    logger = console
+  }: {
+    playlistId: number
+    playlistName: string
+    trackIds: number[]
+    description: string
+    isAlbum: boolean
+    isPrivate: boolean
+    coverArt: string
+    logger: Console
+  }): Promise<{ blockHash: string; blockNumber: number; playlistId: number }> {
+    try {
+      const userId: number = parseInt(this.userStateManager.getCurrentUserId())
+      const updateAction = Action.UPDATE
+      const entityType = 'Playlist'
+      this.REQUIRES(Services.CREATOR_NODE)
+      let dirCID;
+      if (coverArt) {
+        const updatedPlaylistImage = await this.creatorNode.uploadImage(
+          coverArt,
+          true // square
+        )
+        dirCID = updatedPlaylistImage.dirCID
+      }
+
+      const playlist = (await this.discoveryProvider.getPlaylists(1, 0, [playlistId]))[0]
+
+      const metadata = {
+        action: updateAction, // why include action here?
```

Comment:
Reason for adding the action was so that we could easily tell what the intent of this metadata was in the case we had to look it up independently for some reason - open to changing this however